walkthroughs/howto-ingress-enhancements/README.md

```diff
@@ -0,0 +1,437 @@
+# Testing Header/Hostname Matching and Path/Prefix Rewrites
+
+## Concepts
+
+In this walkthrough we'll be using the following set up to test out Header/Hostname Matching and Path/Prefix Rewrites.
+
+![System Diagram](./howto-ingress-enhancements.png "System Diagram")
+
+
+- **GatewayRoute:** Gateway Routes allows specifying routing conditions that match the incoming request and determines the Virtual Service to redirect the request to. These conditions are specified as match conditions (`prefix` , `path`, `queryParameters`, `hostname` for HTTP/HTTP2 routes and `serviceName` and `hostname` for GRPC). A sample spec for the GatewayRoute is as follows:
+
+	```json
+	{
+    "spec": {
+        "httpRoute" : {
+            "match" : {
+                "prefix" : "/red",
+                "queryParameters" : [{
+                    "name" : "color",
+                    "value" : {
+                      "exact" : "red"
+                    }
+                 }],
+                "hostname" : {
+                   "exact" : "www.example.com"
+                },
+                "headers" : [{
+                   "name" : "CACHE_CONTROL",
+                   "match" :  {
+                      "exact" : "no-cache" 
+                   }
+                }]
+            },
+            "action" : {
+                "target" : {
+                    "virtualService": {
+                        "virtualServiceName": $VIRTUALSERVICE_NAME
+                    }
+                }
+            }
+        }
+    }
+	}
+	```
+	A matched request by a gateway route is rewritten to the target Virtual Service's `hostname` and the matched prefix is rewritten to `/`, by default, or when default Prefix rewrite is `Enabled`. 
+	Alternatively, you can specify a custom prefix to rewrite the matched prefix to, as well as specify configuration for matching/rewriting based on paths.
+	Depending on how you configure your Virtual Service, it could then rely on a Virtual Router to route the request to different virtual nodes, based on specific prefixes or headers.
+	
+- **Routes**: A route is associated with a virtual router. The route is used to match requests for the virtual router and to distribute traffic to its associated virtual nodes. If a route matches a request, it can distribute traffic to one or more target virtual nodes. In this walkthrough, we will look at routes matching on `path`, `prefix`, `queryParameters` and `headers` in HTTP Routes. 
+ 
+    A sample spec for Route is:
+    
+    ```
```

Comment:
nit: The formatting is off for these json files. Could you annotate this with json so that it is consistent? Applicable to all specs.

Reply:
Sure, this one was missed. Added.